lib/src/tools/dart_function_tool.dart

```diff
@@ -0,0 +1,68 @@
+import 'dart:async';
+
+import 'package:io/io.dart';
+import 'package:logging/logging.dart';
+
+import '../dart_dev_tool.dart';
+import '../utils/assert_no_positional_args_nor_args_after_separator.dart';
+
+/// A utility class designed to make it simple to create a [DevTool] from just a
+/// Dart function.
+///
+/// To create a [DartFunctionTool], all that is needed is a function of type:
+///     FutureOr<int> Function(DartFunctionToolContext context)
+///
+/// The `DartFunctionToolContext context` contains info about the context in
+/// which the tool is being run, which can be used in the function. Currently,
+/// it only contains one piece if info: whether or not "verbose" mode is
+/// enabled.
+///
+///     import 'package:dart_dev/dart_dev.dart';
+///
+///     int helloWorld(DartFunctionToolContext context) {
+///       print('Hello world!');
+///       return 0;
```

Comment:
Shell conventions here, right? Should we add a constant for "success" to make it more obvious? Kind of a nit

Reply:
Yeah, there is an `ExitCode` constants class available in `pkg:io` that we could use here.

Reply:
Updated